src/query/storages/system/src/virtual_columns_table.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use databend_common_catalog::catalog::CATALOG_DEFAULT;
+use databend_common_catalog::plan::PushDownInfo;
+use databend_common_catalog::table::Table;
+use databend_common_exception::Result;
+use databend_common_expression::types::StringType;
+use databend_common_expression::types::TimestampType;
+use databend_common_expression::DataBlock;
+use databend_common_expression::FromData;
+use databend_common_expression::TableDataType;
+use databend_common_expression::TableField;
+use databend_common_expression::TableSchemaRefExt;
+use databend_common_meta_app::schema::ListVirtualColumnsReq;
+use databend_common_meta_app::schema::TableIdent;
+use databend_common_meta_app::schema::TableInfo;
+use databend_common_meta_app::schema::TableMeta;
+use databend_common_meta_app::schema::VirtualColumnMeta;
+use databend_common_meta_types::MetaId;
+use databend_common_storages_fuse::TableContext;
+
+use crate::columns_table::dump_tables;
+use crate::table::AsyncOneBlockSystemTable;
+use crate::table::AsyncSystemTable;
+
+pub struct VirtualColumnsTable {
+    table_info: TableInfo,
+}
+
+#[async_trait::async_trait]
+impl AsyncSystemTable for VirtualColumnsTable {
+    const NAME: &'static str = "system.virtual_columns";
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn get_full_data(
```

Comment:
Seems we don't allow  `database` or `tables` predications ?

`get_full_data` maybe overhead for queries with filter.